src/v/utils/functional.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "utils/concepts-enabled.h"
+
+#include <functional>
+#include <optional>
+namespace reduce {
+struct push_back {
+    /// TODO(rob) introduce a c++ concept here
+    template<typename vec_like>
+    vec_like operator()(vec_like acc, typename vec_like::value_type t) {
+        acc.push_back(std::move(t));
+        return std::move(acc);
```

Comment:
not sure how this compiles, because in a branch off of this branch, i am seeing this warning (treated as an error)
```
../../../src/v/utils/functional.h:34:29: error: redundant move in return statement [-Werror=redundant-move]
   34 |         return std::move(acc);
```

Reply:
* in gcc release mode

Reply:
Actually its not copy elision, the explanation is very interesting. In this article control-f for the section heading "-Wredundant-move"

https://developers.redhat.com/blog/2019/04/12/understanding-when-not-to-stdmove-in-c/